components/html-block/html-block.js

```diff
@@ -210,51 +209,34 @@ class HtmlBlock extends LitElement {
 		});
 	}
 
-	connectedCallback() {
-		super.connectedCallback();
-
-		if (!this._contentObserver || this.noDeferredRendering) return;
-
-		const slot = this.shadowRoot.querySelector('slot');
-		if (slot) {
-			const slottedNodes = slot.assignedNodes({ flatten: true });
-			this._hasSlottedContent = this._hasSlottedElements(slottedNodes);
-
-			slottedNodes.forEach(
-				node => this._contentObserver.observe(node, { attributes: true, childList: true, subtree: true })
-			);
-		}
-	}
-
-	disconnectedCallback() {
-		super.disconnectedCallback();
-		if (this._contentObserver) this._contentObserver.disconnect();
-	}
-
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 		this._updateContextKeys();
 	}
 
 	render() {
+		if (this.html && this.noDeferredRendering) {
+			throw new Error('Cannot use html attribute with no-deferred-rendering.');
```

Comment:
I'm not 100% sold on this. This is for sure an invalid use case. The previous behaviour here was to ignore the `html` attribute and prioritize the slot in this case, but that was mostly because while both the `html` and `slot` APIs were valid, the likelihood of mixing these by accident was significantly higher. I'm not sure that's a concern anymore.

Reply:
You may want to include the web component name in the exception. For example:

> \<d2l-html-block\>: "html" attribute is not supported with "no-deferred-rendering"

One thing to watch out for is that properties can get set out-of-order, so even if someone's using this in a valid way, `render()` can still be called such that this is in a temporary invalid state.

To get around this in other components, we call a `_validateProperties()` method whenever `html` or `noDeferredRendering` change in `updated()`. It then sets a 3s timeout and _then_ does the check and throws. [Example from color input](https://github.com/BrightspaceUI/core/blob/main/components/inputs/input-color.js#L402).

Reply:
Ah yes, good points. I'll make these changes.

Reply:
I also stole your error message because I like it better. 😆